Gordon360/Services/MembershipRequestService.cs

```diff
@@ -169,9 +169,9 @@ public async Task<RequestView> DeleteAsync(int requestID)
         /// </summary>
         /// <param name="requestID">The membership request id</param>
         /// <returns>The matching RequestView</returns>
-        public RequestView Get(int requestID)
+        public RequestView? Get(int requestID)
         {
-            var request = Get(requestID);
+            var request = _context.RequestView.FirstOrDefault(rv => rv.RequestID == requestID);
 
             if (request == null)
             {
```

Comment:
Technically, this never actually returns null. Instead, it throws an exception if none is found. I would rather it return null so that the caller can decide how to handle that null case instead, but that's just my opinion on exceptions.

I'm guessing, since RequestView is a view in SQL, you can't use `_context.RequestView.Find` here? If not, FirstOrDefault works as well. Find is just convenient in the places that it does work, because it bypasses the database entirely when the entity is already in-memory.

Reply:
I don't think marking it as nullable is the best idea here. That gets rid of the (misguided) warning by giving incorrect type information to the caller of the method. That means that, even though this method never returns null, the caller still has to handle the null case - and they may expect that it returns null instead of throwing an error.

If you want to preserve the behavior of throwing an error when the request is not found (which I still don't love - actually returning null seems like the obvious best solution to me), here's one way you could do that:
```csharp
RequestView? query = _context.RequestView.FirstOrDefault(...);

if (!query is RequestView request) throw new ResourceNotFoundException() ...;

return request;
```

This way, since we're rebinding the result to a non-nullable var `request`, Visual Studio (hopefully) recognizes that we are handling the null case and never actually return null.